v0/destinations/pipedrive/transform.js

```diff
@@ -0,0 +1,416 @@
+   * update org_id field for that person
+   */
+  const response = defaultRequestConfig();
+  response.body.JSON = {
+    org_id: org.id
+  };
+  response.method = defaultPutRequestConfig.requestMethod;
+  response.endpoint = `${PERSONS_ENDPOINT}/${person.id}`;
+  response.params = {
+    api_token: destination.Config.api_token
+  };
+  response.headers = {
+    "Content-Type": "application/json",
+    Accept: "application/json"
+  };
+
+  return response;
+};
+
+/**
+ * Merges two Persons
+ * @param {*} message
+ * @param {*} category
+ * @param {*} destination
+ * @returns
+ */
+const aliasResponseBuilder = async (message, category, destination) => {
+  /**
+   * merge previous Id to userId
+   * merge id to merge_with_id
+   * destination payload structure: { "merge_with_id": "userId"}
+   */
+  const previousId = getValueFromMessage(message, [
+    "previousId",
+    "traits.previousId",
+    "context.traits.previousId"
+  ]);
+  if (!previousId) throw new Error("error: cannot merge without previousId");
+
+  const userId = getFieldValueOrThrowError(
+    message,
+    "userId",
+    new Error("error: cannot merge without userId")
+  );
+
+  /**
+   * Need to extract the pipedrive side integer id for the provided userId
+   * and previous Id, i.e mapping Rudder side userId to pipedrive integer id
+   */
+  const prevPerson = await searchPersonByCustomId(previousId, destination);
+  if (!prevPerson) throw new Error("person not found. cannot merge");
+
+  const currPerson = await searchPersonByCustomId(userId, destination);
+  if (!currPerson) throw new Error("person not found. cannot merge");
+
+  let updatePayload = constructPayload(message, MAPPING_CONFIG[category.name]);
+  updatePayload = extractCustomFields(
+    message,
+    updatePayload,
+    ["traits", "context.traits"],
+    PIPEDRIVE_IDENTIFY_EXCLUSION
+  );
+
+  updatePayload = renameCustomFields(updatePayload, destination.Config.fieldsMap, "person");
+  updatePayload = removeUndefinedAndNullValues(updatePayload);
+
+  /**
+   * if traits is not empty, update the current person first
+   * and then call the merge endpoint
+   */
+  if (Object.keys(updatePayload).length !== 0)
+    updatePerson(currPerson.id, updatePayload, destination);
+
+  const response = defaultRequestConfig();
+  response.method = defaultPutRequestConfig.requestMethod;
+  response.headers = {
+    "Content-Type": "application/json",
+    Accept: "application/json"
+  };
+  response.body.JSON = {
+    merge_with_id: currPerson.id
+  };
+  response.endpoint = getMergeEndpoint(prevPerson.id);
+  response.params = {
+    api_token: destination.Config.api_token
+  };
+
+  return response;
+};
+
+/**
+ * Creates a lead and links it to user.
+ * Supports Ecom Events: "product viewed", "order completed"
+ * @param {*} message
+ * @param {*} category
+ * @param {*} destination
+ * @returns
+ */
+const trackResponseBuilder = async (message, category, destination) => {
+  const userId = getFieldValueOrThrowError(
```

Comment:
do this and the person call only for leads and not product as we are not sending any person specific info to /products?